earth_enterprise/src/fusion/autoingest/sysman/khSystemManager.cpp

```diff
@@ -72,20 +73,43 @@ khSystemManager::SetWantExit(void)
   }
 }
 
+// when SIGHUP is given, reload systemrc
+void systemrc_reload()
+{
+    Systemrc systemrc;
+    LoadSystemrc(systemrc);
+    uint32 logLevel = systemrc.logLevel;
+    assert(logLevel >= 0 && logLevel <= 7);
```

Comment:
I don't think an assert is the right way to handle cases where a user's input might be wrong. I think a better option would be to check the log level in LoadSystemrc. If the level is invalid you can set it to a valid level and print a warning.